docs/dev/reports/README.md

```diff
@@ -0,0 +1,469 @@
+A2(Admission report - ephemeral) -->|watch| AC(Admission reports controller)
+A3(Admission report - ephemeral) -->|watch| AC(Admission reports controller)
+AC(Admission reports controller) -->|create/update| A(Admission report - aggregated)
+AC(Admission reports controller) -->|delete| D1(Admission report - ephemeral)
+AC(Admission reports controller) -->|delete| D2(Admission report - ephemeral)
+AC(Admission reports controller) -->|delete| D3(Admission report - ephemeral)
+A(Admission report - aggregated) -->|watch| C(Policy report controller)
+B1(Background scan report) -->|watch| C(Policy report controller)
+B2(Background scan report) -->|watch| C(Policy report controller)
+B3(Background scan report) -->|watch| C(Policy report controller)
+C(Policy report controller) -->|create/update| R1(Policy report)
+C(Policy report controller) -->|create/update| R2(Policy report)
+C(Policy report controller) -->|create/update| R3(Policy report)
+C(Policy report controller) -->|create/update| R4(Policy report)
+```
+
+Not shown in this diagram:
+- background scan controller watches resources/policies and maintains an up to date background scan report
+
+## Storage considerations
+
+The system stores everything in etcd, admission reports (aggregated and short lived ones), background scan reports, and policy reports/cluster policy reports.
+
+### Admission reports and background scan reports
+
+If we omit the short lived admission reports, this usually means 2 additionnal resources per managed resource:
+- one aggregated admission report
+- one background scan report
+
+Short lived admission reports are ephemeral in nature, as long as they cleaned up correctly they shouldn't impact storage too much.
+If for some reason Kyverno fails to cleanup those reports fast enough it can become a severe issue though.
+
+Of course not all resources will have background scan reports, some policies can have `background` disabled but you get the picture.
+One thing to note here is that it doesn't vary with the number of policies in the cluster. The number of policies will have an impact on the report size though, but not on the number of reports.
+
+Still, admission and background scan reports are usually small sized because they are for a single resource.
+
+### Policy reports and cluster policy reports
+
+For policy reports it's different, the number of policy reports will depend on the number of policies and the number of namespaces:
+- one report per policy and per namespace for namespaced resources (10 policies and 20 namespaces -> 200 reports)
+- one report per policy for clustered resources (10 policies -> 10 reports)
+
+Policy reports and cluster policy reports are usually much larger, depending on the number of resources in the cluster/namespace.
+Sometimes they are too big to be stored in etcd and we split them into multiple reports containing a configurable number entries (`1000` by default).
```

Comment:
do we automatically split based on the entry count?

Reply:
Yes, once a given report reaches 1,000 results, a second report is created.